File: Donggyo/day03/Q04_귤고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        int[] counts = new int[10000001];
        
        for(int t:tangerine){
            counts[t]++;
        }
        Arrays.sort(counts);
        
        for(int i = counts.length-1; i >= 0; i--){
            k-=counts[i];
            answer++;
            if(k<=0) break;
        }
        return answer;
    }
}

File: Jiho/day03/Q04_귤 고르기.js
function solution(k, tangerine) {
  let answer = 0;
  let counts = {};

  // 개수별 정리
  tangerine.forEach((t) => {
    if (!(t in counts)) counts[t] = 0;
    counts[t]++;
  });

  // 개수 많은 순 정렬
  counts = Object.entries(counts)
    .sort((a, b) => b[1] - a[1])
    .map((v) => v[1]);

  // answer
  for (let i = 0; i < counts.length; i++) {
    k -= counts[i];
    answer++;
    if (k <= 0) break;
  }

  return answer;
}
